addons/web/static/src/search/control_panel/control_panel.js

```diff
@@ -31,14 +65,34 @@ export class ControlPanel extends Component {
         this.pagerProps = this.env.config.pagerProps
             ? useState(this.env.config.pagerProps)
             : undefined;
+        this.notificationService = useService("notification");
         this.breadcrumbs = useState(this.env.config.breadcrumbs);
-
+        this.orm = useService("orm");
+        this.dialogService = useService("dialog");
         this.root = useRef("root");
+        this.newActionNameRef = useRef("newActionNameRef");
+        this.isTopBarActionsModifiable = false;
+        /**
+         * The visible topbar actions are unique to each user and to each res_id. The visible actions chosen by the
+         * user are stored in the local storage in a key corresponding to a combination of the actionId, the activeId
+         * and the currrent userId. Each key contains a dict. The keys of the latter are the id of the visible topbar
+         * actions.
+         */
+        this.localStorageKeyActionIdResId = `visibleTopBarActions${
+            this.env.config.parentActionId || ""
+        }+${this.env.searchModel?.globalContext.active_id || ""}+${user.userId}`;
 
         this.state = useState({
+            showTopBar: this.env.config.fromParentAction || false,
             showSearchBar: false,
             showMobileSearch: false,
             showViewSwitcher: false,
+            topBarActions: this.env.config.topBarActions,
+            newActionShared: false,
+            newActionName: `${this.getCurrentTopBarAction?.name} Custom`,
+            visibleTopBarActions:
+                JSON.parse(localStorage.getItem(this.localStorageKeyActionIdResId)) || {},
+            currentTopBarAction: this.getCurrentTopBarAction,
```

Comment:
I find it a little weird that all those info are on the same level as other `show...` keys.
Is it also reactive if you put them in a dict, with the key `topBarInfos` or something like that?